crypto.go

```diff
@@ -93,7 +93,11 @@ func (i *Identity) ConfigForPeer(remote peer.ID) (*tls.Config, <-chan ic.PubKey)
 			return err
 		}
 		if remote != "" && !remote.MatchesPublicKey(pubKey) {
-			return errors.New("peer IDs don't match")
+			var peerID peer.ID = "(not determined)"
+			if p, err := peer.IDFromPublicKey(pubKey); err == nil {
+				peerID = p
+			}
+			return fmt.Errorf("peer IDs don't match. Expected %s, got %s", remote, peerID)
```

Comment:
nit: "peer IDs don't match: expected %s, got %s" but that's mostly just preference.

Reply:
Fixed.